src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -76,31 +77,112 @@ GEQRF_STRIDED_BATCH_LAUNCHER(std::complex<double>, cusolverDnZgeqrf)
-                 std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "getri_batch");
-}
-void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<double> &a, std::int64_t lda,
-                 std::int64_t stride_a, sycl::buffer<std::int64_t> &ipiv, std::int64_t stride_ipiv,
-                 std::int64_t batch_size, sycl::buffer<double> &scratchpad,
-                 std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "getri_batch");
-}
-void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<std::complex<float>> &a,
-                 std::int64_t lda, std::int64_t stride_a, sycl::buffer<std::int64_t> &ipiv,
-                 std::int64_t stride_ipiv, std::int64_t batch_size,
-                 sycl::buffer<std::complex<float>> &scratchpad, std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "getri_batch");
-}
-void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<std::complex<double>> &a,
-                 std::int64_t lda, std::int64_t stride_a, sycl::buffer<std::int64_t> &ipiv,
-                 std::int64_t stride_ipiv, std::int64_t batch_size,
-                 sycl::buffer<std::complex<double>> &scratchpad, std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "getri_batch");
+template <typename Func, typename T>
+inline void getri_batch(const char *func_name, Func func, sycl::queue &queue, std::int64_t n,
+                        sycl::buffer<T> &a, std::int64_t lda, std::int64_t stride_a,
+                        sycl::buffer<std::int64_t> &ipiv, std::int64_t stride_ipiv,
+                        std::int64_t batch_size, sycl::buffer<T> &scratchpad,
+                        std::int64_t scratchpad_size) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+
+    overflow_check(n, lda, stride_a, stride_ipiv, batch_size, scratchpad_size);
+
+    std::uint64_t ipiv32_size = n * batch_size;
+    sycl::buffer<int> ipiv32(sycl::range<1>{ ipiv32_size });
+    sycl::buffer<int> devInfo{ batch_size };
+
+    queue.submit([&](sycl::handler &cgh) {
+        auto ipiv_acc = sycl::accessor{ ipiv, cgh, sycl::read_only };
+        auto ipiv32_acc = sycl::accessor{ ipiv32, cgh, sycl::write_only };
+        cgh.parallel_for(sycl::range<1>{ static_cast<size_t>(ipiv32_size) },
+                         [=](sycl::id<1> index) {
+                             ipiv32_acc[index] =
+                                 static_cast<int>(ipiv_acc[(index / n) * stride_ipiv + index % n]);
+                         });
+    });
+
+    // getri_batched is contained within cublas, not cusolver.
+    queue.submit([&](sycl::handler &cgh) {
+        using blas::cublas::cublas_error;
+
+        sycl::accessor a_acc{ a, cgh, sycl::read_only };
+        sycl::accessor scratch_acc{ scratchpad, cgh, sycl::write_only };
+        sycl::accessor ipiv32_acc{ ipiv32, cgh };
+        sycl::accessor devInfo_acc{ devInfo, cgh, sycl::write_only };
```

Comment:
Not a major issue, but camel case mixed snake case?

Reply:
For consistency with other parts of the codebase I am going to leave it like this if you think this is OK